include/ruby/internal/error.h

```diff
@@ -50,6 +50,9 @@ typedef enum {
     /** Warning is for experimental features. */
     RB_WARN_CATEGORY_EXPERIMENTAL,
 
+    /** Warning is for performance. */
+    RB_WARN_CATEGORY_PERFORMANCE,
```

Comment:
Shouldn't this also be added to the `RB_WARN_CATEGORY_ALL_BITS` below?

Reply:
Nope, it was specifically asked that this category is not enabled on bare`-W`. `RB_WARN_CATEGORY_ALL_BITS` is what controls that.